src/App.jsx

```diff
@@ -15,32 +15,58 @@ class App extends React.Component {
       cocktails1: [],
       cocktails2: [],
       intersection: [],
+      errorMessage: 'ERROOOOOOOOOOOOOORS',
+      errorShow: false,
     };
   }
 
 setKeywords1 = (keywords1) => this.setState({ keywords1 })
 setKeywords2 = (keywords2) => this.setState({ keywords2 })
 
+manageError = () => {
+  this.setState({ errorShow : !this.state.errorShow });
+}
+
 searchIngredient1 = () => {
-  Axios.get(`https://www.thecocktaildb.com/api/json/v1/1/filter.php?i=${this.state.keywords1}`)
+  
+  Axios.get(`https://www.thecocktaildb.com/api/json/v1/1/filter.php?i=${this.state.keywords1 !== undefined ? this.state.keywords1 : ''}`)
     .then((response) => response.data)
     .then((data) => {
       this.setState({ cocktails1: data.drinks });
     });
 }
 
 searchIngredient2 = () => {
-  Axios.get(`https://www.thecocktaildb.com/api/json/v1/1/filter.php?i=${this.state.keywords2}`)
+  
+  Axios.get(`https://www.thecocktaildb.com/api/json/v1/1/filter.php?i=${this.state.keywords2 !== undefined ? this.state.keywords2 : ''} `)
     .then((response) => response.data)
     .then((data) => {
       this.setState({ cocktails2: data.drinks });
     });
 }
 
 compare = () => {
-  this.setState ({ intersection:  this.state.cocktails1.filter((cocktail1) => this.state.cocktails2.findIndex(cocktail2 => cocktail1.strDrink === cocktail2.strDrink) !== -1)})
+  if (this.state.cocktails1 === undefined || this.state.cocktails2 === undefined) {
+    this.setState ({ errorMessage : 'Il manque un ingrédient' })
+    this.setState ({ errorShow : true })
+  } else {
+    this.setState ({ intersection: [] });
+    this.setState ({ intersection:  this.state.cocktails1.filter((cocktail1) => this.state.cocktails2.findIndex(cocktail2 => cocktail1.strDrink === cocktail2.strDrink) !== -1)})
+    if (this.state.intersection.length === 0) {
+      this.setState ({ errorMessage :'aucun cocktail ne contient ces ingrédients'});
+      this.setState ({ errorShow : true })
+
+    }
+  }
 }
 
+fullConsoleLog = () => {
+  console.log(this.state.keywords1);
+  console.log(this.state.cocktails1);
+  console.log(this.state.keywords2);
+  console.log(this.state.cocktails2);
+  console.log(this.state.intersection)
+}
```

Comment:
use react dev tools to see the state ;)

Reply:
oh, thats just my tool that i have to remove but forget it

Reply:
Done ;)